### PR TITLE
[Echo] Take first value found rather than last

### PR DESCRIPTION
**Pull request by dracos**
For https://github.com/mysociety/fixmystreet/pull/3348
Also includes the parent change from another branch so it's hopefully easier to merge that in later.

### CODE REVIEW

**Review by struan — APPROVED** [review]
👍

File: perllib/Open311/Endpoint/Integration/Echo.pm [hunk 1]
```diff
@@ -115,15 +115,37 @@ sub post_service_request {
         my $row = { id => $type->{Id} };
         $row->{value} = $self->check_for_data_value($type->{Name}, $args, $request);
 
+        my %extra;
+        my $extra = 0;
```
Comment (struan) [inline perllib/Open311/Endpoint/Integration/Echo.pm:119]: might be clearer if this was called `$extra_count`

File: t/open311/endpoint/echo.t [hunk 2]
```diff
@@ -80,6 +87,20 @@ $soap_lite->mock(call => sub {
...
+                @children = ${$notes[0]->value}->value->value;
+                @child = ${$children[0]->value}->value;
+                is $child[0]->value, 1005;
+                is $child[1]->value, 2;
+                @child = ${$children[1]->value}->value;
+                is $child[0]->value, 1006;
+                is $child[1]->value, 1;
+                @child = ${$children[2]->value}->value;
+                is $child[0]->value, 1007;
+                is $child[1]->value, 12;
```
Comment (struan) [inline t/open311/endpoint/echo.t:102]: should this not also check there's a second event here with a reason of 7?

**Review by struan — CHANGES_REQUESTED** [review]
one variable name tweak and a bit more testing.

File: t/open311/endpoint/echo.t (hunk 2 above)
Comment (dracos) [reply t/open311/endpoint/echo.t:102]: It did that above when it runs the same test as the non-multi request, then this is extra to test the second one on top.

Comment (struan) [reply t/open311/endpoint/echo.t:102]: ah yes, so it does.

**Pull request merged**